geomet_data_registry/util.py

```diff
@@ -26,6 +26,47 @@
 DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
 
 
+class VRTDataset:
+    """
+    Object used to create VRTs. The object can be passed a list of ordered
+    variable names to sort the provided file paths on and create a VRT with the
+    bands in the specified list order (ex. UU layers where the U data needs
+    to be the first band and the V data the second band).
+    """
+    def __init__(self, filepaths, raster_x_size=None, raster_y_size=None,
+                 bands_order=None):
+        self.vrt_dataset = [
+            '<VRTDataset rasterXSize="{}" rasterYSize="{}" bands="{}">'.format(
+                raster_x_size, raster_y_size, len(filepaths)),
+            '</VRTDataset>'
+        ]
+        self.vrt_raster_band_template = '''
+<VRTRasterBand dataType="Byte" band="{}">
+    <ComplexSource>
+        <SourceFilename>{}</SourceFilename>
+        <SourceBand>1</SourceBand>
+        <ScaleOffset>0.0</ScaleOffset>
+        <ScaleRatio>1.0</ScaleRatio>
+    </ComplexSource>
+</VRTRasterBand>
+        '''.replace('\n', '')
+        self.filepaths = filepaths
+        self.bands_order = bands_order
+
+    def build(self):
+        if self.bands_order:
+            self.filepaths = sorted(self.filepaths, key=lambda fp:
+                                    self.sort_bands(fp))
+        for index, filepath in enumerate(self.filepaths, start=1):
+            self.vrt_dataset.insert(-1, self.vrt_raster_band_template.format(
+                index, filepath))
+        return ''.join(self.vrt_dataset)
+
+    def sort_bands(self, filepath):
+        return [idx for idx, value in enumerate(self.bands_order) if value
```

Comment:
- Log what's going on here?
- how does the else condition evaluate?

Reply:
The sort bands method now returns a `VRTDatasetError` if none of the `self.bands_order` values can be found in the provided filepath, meaning that the filepaths cannot be properly sorted prior to the VRT construction.

I also added a log debug message indicating that the filepath is being checked against the provided bands order.